src/screens/Settings/hooks/useSettings.js

```diff
@@ -34,12 +34,12 @@ const useSettings = () => {
   const logout = async () => {
     try {
       await OneSignalUtil.removeAllSubscribedTags();
-      OneSignalUtil.removeExternalId();
-      StorageUtils.clearAll();
-
       const fcmToken = await messaging().getToken();
       await removeFcmToken(fcmToken);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `messaging().getToken()` function retrieves the current FCM token for this device, which may not be the same as the one that was used when the user logged in. If the token has changed since the user logged in, you might end up removing a token that is currently not associated with the user. This could lead to unexpected behavior.

To ensure that you are removing the correct FCM token, you should store the FCM token at the time of login and then remove that specific token during logout. 

Here's an example of how you can do this:

```diff
+  const fcmToken = await StorageUtils.getFcmToken();
   await removeFcmToken(fcmToken);
```

In the above code, `StorageUtils.getFcmToken()` is a hypothetical function that retrieves the FCM token from storage. You would need to implement this function, and also make sure to store the FCM token in storage at the time of login.


<!-- This is an auto-generated comment by OSS CodeRabbit -->